src/fr/flamescansfr/src/eu/kanade/tachiyomi/extension/fr/flamescansfr/LegacyScans.kt

```diff
@@ -1,9 +1,202 @@
 package eu.kanade.tachiyomi.extension.fr.flamescansfr
 
-import eu.kanade.tachiyomi.multisrc.mangathemesia.MangaThemesia
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
 import java.text.SimpleDateFormat
 import java.util.Locale
 
-class LegacyScans : MangaThemesia("Legacy Scans", "https://legacy-scans.com", "fr", dateFormat = SimpleDateFormat("MMMM d, yyyy", Locale.FRENCH)) {
-    override val id = 8947802555328550956
+class LegacyScans : HttpSource() {
+    override val name: String = "Legacy Scans"
+
+    override val lang: String = "fr"
+
+    override val baseUrl: String = "https://legacy-scans.com"
+
+    override val supportsLatest: Boolean = true
+
+    override val versionId: Int = 2
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .build()
+
+    private val json: Json by injectLazy()
+
+    override fun popularMangaRequest(page: Int) = GET("$apiUrl/misc/views/all", headers)
+
+    override fun popularMangaParse(response: Response) =
+        mangasPageParse(response.parseAs<List<MangaDto>>()).copy(hasNextPage = false)
```

Comment:
pass an optional parameter to `mangasPageParse` to set `hasNextPage` so copying can be avoided